src/NumericalAlgorithms/Spectral/Chebyshev.cpp

```diff
@@ -0,0 +1,118 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+#include <cmath>
+#include <cstddef>
+// IWYU pragma: no_include <type_traits>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+namespace Spectral {
+
+// Algorithms to compute Chebyshev basis functions
+// These functions specialize the templates declared in `Spectral.cpp`.
+
+/// \cond
+template <Basis>
+DataVector compute_basis_function_values(size_t, const DataVector&) noexcept;
+template <>
+DataVector compute_basis_function_values<Basis::Chebyshev>(
+    const size_t k, const DataVector& x) noexcept {
+  // Algorithm 21 from Kopriva's book, p. 60
+  switch (k) {
+    case 0:
+      return DataVector(x.size(), 1.);
+    case 1:
+      return x;
+    default:
+      // These values can be computed either through recursion
+      // (implemented here), or analytically as `cos(k * acos(x))`.
+      // Since the trigonometric form is expensive to compute it is useful only
```

Comment:
The trig method will never be useful for us - it's only faster than recurrence when N is many dozens, and the recurrence is accurate for N in the many thousands. I think it's fine for us not to bother with an assert here.

Reply:
I'm now confused. The C++ comment makes it sound like we don't do trig because it's expensive but is also more accurate. I'm interpreting your response as saying the recursion is more accurate, which confuses me.

I still think if we cache these values we should just use the trig form since any "cost" is basically zero anyway.

Reply:
I'm not sure where the claim that the trig form is more accurate comes from. Trefethen has a whole book and software package for computing with Chebyshev polynomials and uses recurrence to compute them. The only place I've seen the trig form used is in Boyd's book, which is over 30 years old. Bottom line: use recurrences, they're faster and more accurate. Anyway, we're "never" going to use Chebyshev polynomials for DG.

Reply:
Ah okay, then the comment needs to be update so that it is actually correct. Thanks for the clarification! :)

Reply:
The comment is actually correct. There‘s also a reference to more information.

Reply:
I don't see how it wouldn't be more accurate for this case, or at least just as accurate. The comment is very misleading and implying there are accuracy reasons for switching from recursion. It definitely needs to be corrected.

Reply:
I don‘t understand, the comment only says that the trigonometric form is more expensive.

Reply:
No, it gives a reason not to use it under certain conditions, not explaining why those conditions are imposed and when, leaving it to the reader to assume what is meant.

Reply:
I looked at the reference (Kopriva). He says the crossover point in terms of speed is around N=70. So let's stick wtih recursion and leave the comment as is.